### PR TITLE
HPCC-25324 Optimize WuDetails with large number of activties per subgraph

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith please review.

**Comment by richardkchapman** [comment]
SHould this target 7.8.x ?

**Comment by ghalliday** [comment]
@richardkchapman yes, I think worthwhile because of the benefit getting graph information for a workunit.  Needs some careful checking though.

**Comment by richardkchapman** [comment]
Typo in commit title

**Comment by jakesmith** [comment]
NB: smoketest failure is for same reason as : https://github.com/hpcc-systems/HPCC-Platform/pull/14555#issuecomment-776017014

**Review by jakesmith — APPROVED** [review]
one trivial comment, but looks good to me.

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -1575,20 +1575,13 @@ class GraphScopeIterator : public CInterfaceOf<IConstWUScopeIterator>
...
+#ifdef _DEBUG
+                assertex(treeIters.tos().query().getPropInt("att[@name='_kind']/@value") == TAKsubgraph);
                 unsigned numIters = treeIters.ordinality();
-                //This should really be implemented by a filter on the node - but it would require _kind/_parentActivity to move to the node tag
-                if (treeIters.tos().query().getPropInt("att[@name='_kind']/@value") != TAKsubgraph)
-                {
-                    state = SChildGraphNext;
-                    break;
-                }
                 unsigned parentActivityId = treeIters.item(numIters-2).query().getPropInt("@id");
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply common/workunit/workunit.cpp:1581]: item() would assert if so, I think?

[the remark this answers is not in the document]
Comment (jakesmith) [reply common/workunit/workunit.cpp:1581]: you're right. it's fine as it is then.

**Comment by ghalliday** [comment]
I checked the code again and can't see any issues.

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**